WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -494,10 +503,34 @@ private extension CollectOrderPaymentUseCase {
             onCompleted()
         }
     }
+    /// Allow merchants to print or email backend-generated receipts.
+    /// The alerts presenter can be simplified once we remove legacy receipts: https://github.com/woocommerce/woocommerce-ios/issues/11897
+    ///
+    func presentBackendReceiptAlert(alertProvider paymentAlerts: CardReaderTransactionAlertsProviding, onCompleted: @escaping () -> ()) {
+        // Present receipt alert
+        alertsPresenter.present(viewModel: paymentAlerts.success(printReceipt: { [weak self] in
```

Comment:
The print and email closures look identical here... will they stay that way? If so, it might be better to just define it once in a variable and pass it in.

Reply:
They will remain until we remove the older way to generate receipts, yes, at that time can remove one of the closures to make it simpler. Defining it into a variable and passing it in seems cleaner, done here: https://github.com/woocommerce/woocommerce-ios/pull/11896/commits/40873e15e92faeaff8a142995ff8615fa45aa0fd

At the moment I'm declaring it internally on `presentBackendReceiptAlert()` for simplicity, but I'll check into passing it from the outside.

Reply:
Internally is fine, thanks :)